src/Optimization/hiopAlgFilterIPM.hpp

```diff
@@ -339,8 +345,74 @@ class hiopAlgFilterIPMQuasiNewton : public hiopAlgFilterIPMBase
   virtual ~hiopAlgFilterIPMQuasiNewton();
 
   virtual hiopSolveStatus run();
+
+  // note that checkpointing is only available with a axom-enabled build
+#ifdef HIOP_USE_AXOM
+  /**
+   * @brief Save state of HiOp algorithm to a sidre data store.
+   * @param data_store a pointer to DataStore
+   *
+   * @details 
+   * A new sidre::group "hiop solver" is created within data_store. Then a sidre::View 
+   * is created within the group for each of the algorithm's states and the states are
+   * copied in the corresponding views. 
+   */
+  virtual void save_state_to_data_store(::axom::sidre::DataStore* data_store);
+
+  /**
+   * @brief Load state of HiOp algorithm from a sidre data store.
+   * @param data_store a pointer to DataStore
+   *
+   * @details 
+   * Copies views from the data store sidre::group (named "hiop solver") to HiOp algorithm's
+   * state variables. The group should be created by first calling save_state_to_data_store 
+   * for a problem/NLP of the same sizes as the problem for which load_state_from_data_store
+   * is called. This ensures the views have the names and sizes expected by this method. 
+   * Otherwise a std::runtime_error exception is thrown.
+   */ 
+  virtual void load_state_from_data_store(const ::axom::sidre::DataStore* data_store);
```

Comment:
```suggestion
  virtual void load_state_from_data_store(const ::axom::sidre::DataStore &data_store);
```
Again, I think reference makes the meaning clear and shows that `nullptr` isn't a valid incoming value. It avoids the uncertainty that comes with a raw pointer.